restclient/src/main/java/com/box/l10n/mojito/rest/resttemplate/ProxyOutboundRequestInterceptor.java

```diff
@@ -0,0 +1,83 @@
+package com.box.l10n.mojito.rest.resttemplate;
+
+import java.io.IOException;
+import java.net.URI;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Component
+public class ProxyOutboundRequestInterceptor implements ClientHttpRequestInterceptor {
```

Comment:
This should be configurable as not all deployments will need it, only those utilising a service proxy